components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/ThemeManagerHostObject.java

```diff
@@ -0,0 +1,437 @@
+        } catch (IOException e) {
+            handleException("Could not delete directory :" + dir.getPath());
+        }
+    }
+
+    /**
+     * Check given theme exists.
+     *
+     * @param path File Path
+     * @return true if exists else false
+     */
+    private static boolean isThemeExists(String path) {
+        File dir = new File(path);
+        if (log.isDebugEnabled()) {
+            String msg = "Custom theme found status : " + dir.exists() + " in path : " + path;
+            log.debug(msg);
+        }
+        return dir.exists();
+    }
+
+    /**
+     * Deploy the uploaded custom theme to the correct custom theme directory. Files with unsupported extensions will be
+     * omitted.
+     *
+     * @param themeFile      Theme File in zip format
+     * @param tenant         Tenant Domain
+     * @param themeType      Theme Type (Default ,<assetType> e.g webapp)
+     * @param whitelistedExt Whitelisted file extensions
+     * @throws AppManagementException
+     */
+    private static void deployCustomTheme(FileHostObject themeFile, String tenant, String themeType,
+                                          Set<String> whitelistedExt) throws AppManagementException {
+
+        if (log.isDebugEnabled()) {
+            String msg = String.format("Deploy custom theme of type :%1s for tenant :%2s", themeType, tenant);
+            log.debug(msg);
+        }
+        ZipInputStream zis = null;
+        byte[] buffer = new byte[1024];
+
+        //check store theme directory exists
+        File themeDir = new File(getStoreThemePath());
+        if (!themeDir.exists()) {
+            String msg = "Could not found directory :" + themeDir;
+            handleException(msg);
+        }
+
+        String themePath = null;
+        themePath = getCustomThemePath(tenant, themeType);
+        InputStream zipInputStream = null;
+        try {
+            zipInputStream = themeFile.getInputStream();
+        } catch (ScriptException e) {
+            handleException("Error occurred while deploying custom theme file", e);
+        }
+
+        try {
+            if (log.isDebugEnabled()) {
+                String msg = String.format("Create custom theme dir :%s", themePath);
+                log.debug(msg);
+            }
+            //create output directory if it is not exists
+            File dir = new File(themePath);
+            if (!dir.exists()) {
+                if (!dir.mkdirs()) {
+                    handleException("Unable to create tenant custom directory :" + dir);
+                }
+            }
+
+            if (log.isDebugEnabled()) {
+                String msg = "Get zip file content and deploy";
+                log.debug(msg);
+            }
+            //get the zip file content
+            zis = new ZipInputStream(zipInputStream);
+            //get the zipped file list entry
+            ZipEntry ze = zis.getNextEntry();
+            String ext = null;
+
+            while (ze != null) {
```

Comment:
Refactor this logic, refer [this](https://github.com/wso2/carbon-uuf/blob/master/components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/io/util/ZipArtifactHandler.java#L92-L119) snippet.